Translation of useInsertionEffect (#516)
reactjs/id.react.dev: src/content/reference/react/useInsertionEffect.md

<Pitfall>

> `useInsertionEffect` digunakan untuk pengguna library CSS-in-JS. Kecuali jika Anda sedang mengerjakan sebuah pustaka CSS-in-JS dan butuh sebuah tempat untuk menyisipkan style, Sudah pasti Anda memilih [`useEffect`](/reference/react/useEffect) atau [`useLayoutEffect`](/reference/react/useLayoutEffect) daripada sebelumya.

Reviewer: style bisa pake italic.

```suggestion
`useInsertionEffect` digunakan untuk pengguna library CSS-in-JS. Kecuali jika Anda sedang mengerjakan sebuah pustaka CSS-in-JS dan butuh sebuah tempat untuk menyisipkan *style*, Anda mungkin lebih menginginkan [`useEffect`](/reference/react/useEffect) atau [`useLayoutEffect`](/reference/react/useLayoutEffect).
```

Reviewer: memperbaiki kalimat sedikit sekalian

<Pitfall>

> `useInsertionEffect` digunakan untuk pengguna library CSS-in-JS. Kecuali jika Anda sedang mengerjakan sebuah pustaka CSS-in-JS dan butuh sebuah tempat untuk menyisipkan *style*, Anda mungkin lebih menginginka [`useEffect`](/reference/react/useEffect) atau [`useLayoutEffect`](/reference/react/useLayoutEffect).

Reviewer: Typo

```suggestion
`useInsertionEffect` digunakan untuk pengguna library CSS-in-JS. Kecuali jika Anda sedang mengerjakan sebuah pustaka CSS-in-JS dan butuh sebuah tempat untuk menyisipkan *style*, Anda mungkin lebih menginginkan [`useEffect`](/reference/react/useEffect) atau [`useLayoutEffect`](/reference/react/useLayoutEffect).
```

Author: Okay mas updated

}
```

> [Lihat contoh lain dibawah ini.](#usage)

Reviewer 2: ```suggestion
[Lihat contoh lain di bawah ini.](#usage)
```

dibawah -> di bawah

#### Parameter {/*parameters*/}

> * `setup`: Fungsi berisi logika Efek Anda. Fungsi *setup* juga dapat secara opsional mengembalikan fungsi *pembersihan* (*cleanup*). Sebelum komponen ditambahkan ke DOM, React akan menjalankan fungsi *setup*. Setelah setiap *re-render* dengan dependensi yang berubah, React akan terlebih dahulu menjalankan fungsi *pembersihan* (*cleanup*) (jika Anda memberikannya) dengan nilai lama. Selanjutnya, React akan menjalankan fungsi *setup* dengan nilai baru. Sebelum komponen dihapus dari DOM, React akan menjalankan fungsi *pembersihan* (*cleanup*).

Reviewer: re-render = `*render* ulang`

#### Returns {/*returns*/}

`useInsertionEffect` mengembalikan `undefined`.

#### Caveats {/*caveats*/}

* Efek hanya berjalan di sisi klien. Efek tidak berjalan ketika *server rendering*.
* Anda tidak dapat mengupdate *state* dari dalam `useInsertionEffect`.
* Pada saat `useInsertionEffect` berjalan, *refs* belum  terpasang, dan DOM belum diperbarui.

---

## Penggunaan {/*usage*/}

> ### Menyisipkan style dimanis fari pustaka CSS-in-JS {/*injecting-dynamic-styles-from-css-in-js-libraries*/}

Reviewer: typo
```suggestion
### Menyisipkan style dinamis dari pustaka CSS-in-JS {/*injecting-dynamic-styles-from-css-in-js-libraries*/}
```

> ### Menyisipkan *style* dimanis dari pustaka CSS-in-JS {/*injecting-dynamic-styles-from-css-in-js-libraries*/}

Reviewer 2: ```suggestion
### Menyisipkan style dinamis dari pustaka CSS-in-JS {/*injecting-dynamic-styles-from-css-in-js-libraries*/}
```

- dimanis -> dinamis
- Don't italicise loanwords in headings.

.success { color: green; }
```

Beberapa tim lebih memilih untuk menulis style secara langsung di dalam pada kode Javascript daripada menuliskan file CSS terpisah. Biasanya ini memerlukan penggunaan sebuah pustaka CSS-in-JS atau sebuah alat. Terdapat 3 pendekatan umum dalam CSS-in-JS:

1. Ekstraksi statis ke dalam file-file CSS dengan sebuah kompiler

> 2. Style *inline*, e.g. `<div style={{ opacity: 1 }}>`

Reviewer: Style pakai italic

3. Injeksi *runtime* tag `<style>`

Jika kamu menggunakan CC-in-JS, kami merekomendasikan sebuah kombinasi dari dua pendekatan pertama (File CSS untuk style statis, *inline* untuk style dinamis). **Kami tidak merekomendasikan injeksi *runtime* tag `<style>` karena dua alasan:**

1. Injeksi *runtime* memaksa browser untuk menghitung ulang style dengan lebih sering.
2. Injeksi *runtime* bisa lebih lambat jika injeksi terjadi pada waktu yang salah pada *React lifecycle*.

Masalah pertama tidak dapat diselesaikan, tapi `useInsertionEffect` membantu Anda menyelesaikan masalah kedua.

> Panggil `useInsertionEffect` untuk menyertakan sebuah style sebelum terjadinya mutasi DOM:

Reviewer: Style pakai italic

}
```

> [Baca lebih lanjut tentang memperbaharui pustaka CSS-in-JS dengan injeksi *runtime* ke `useInsertionEffect`.](https://github.com/reactwg/react-18/discussions/110)

Reviewer: Memperbarui bukan memperbaharui

<DeepDive>

> #### Bagaimana ini bisa lebih baik daripada menyisipkan style saat render atau useLayoutEffect? {/*how-is-this-better-than-injecting-styles-during-rendering-or-uselayouteffect*/}

Reviewer: style & render pakai italic

> #### Bagaimana ini bisa lebih baik daripada menyisipkan *style* saat *render* atau useLayoutEffect? {/*how-is-this-better-than-injecting-styles-during-rendering-or-uselayouteffect*/}

Reviewer 2: ```suggestion
#### Bagaimana ini bisa lebih baik daripada menyisipkan style saat render atau useLayoutEffect? {/*how-is-this-better-than-injecting-styles-during-rendering-or-uselayouteffect*/}
```

Idem.

> Jika kamu menambahkan style saat rendering dan React sedang memproses [non-blocking update,](/reference/react/useTransition#marking-a-state-update-as-a-non-blocking-transition) browser akan menghitung ulang style setiap *frame* saat me-*render* sebuah pohon (*tree*) komponen, yang dapat **sangat lambat.**

Reviewer: style pakai italic

Reviewer: browser = peramban

> `useInsertionEffect` lebih baik daripada menyertakan style saat [`useLayoutEffect`](/reference/react/useLayoutEffect) atau [`useEffect`](/reference/react/useEffect) karena `useInsertionEffect` memastikan pada saat efek lain berjalan di dalam komponen Anda, tag `<style>` sudah terpasang. Jika tidak, perhitungan tata letak pada efek biasa akan salah karena style yang sudah usang.

Reviewer: style pakai italic
